File: api/callRecordings.js
const express = require('express');
const axios = require('axios');
const jwt = require('jsonwebtoken');

const app = express();
const PORT = 3000;

// Middleware for token-based authentication
app.use((req, res, next) => {
    const token = req.headers['authorization']?.split(' ')[1];
    if (!token) {
        return res.status(403).json({ message: 'No token provided.' });
    }
    jwt.verify(token, 'YOUR_SECRET_KEY', (err, decoded) => {
        if (err) {
            return res.status(401).json({ message: 'Unauthorized.' });
        }
        req.userId = decoded.id;
        next();
    });
});

// Endpoint to fetch call recordings
app.get('/api/callRecordings', async (req, res) => {
    try {
        const response = await axios.get('https://externalapi.com/callRecordings');
        const callRecordings = response.data;

        // Fetch funding info separately
        const fundingInfoPromises = callRecordings.map(recording =>
            axios.get(`https://externalapi.com/fundingInfo/${recording.id}`)
        );

        const fundingInfos = await Promise.all(fundingInfoPromises);

        const structuredResponse = callRecordings.map((recording, index) => ({
            recording,
            funding: fundingInfos[index].data
        }));

        res.json(structuredResponse);
    } catch (error) {
        res.status(500).json({ message: 'Error fetching data.' });
    }
});

app.listen(PORT, () => {
    console.log(`Server running on port ${PORT}`);
});

File: api/call_recordings.py
from flask import Flask, jsonify, request
import requests
from functools import wraps

app = Flask(__name__)

# Mock external API base URL
EXTERNAL_API_URL = 'https://externalapi.com/call_recordings'

# Token-based authentication
API_TOKEN = 'your_secure_token'

# Authentication decorator
def token_required(f):
    @wraps(f)
    def decorated(*args, **kwargs):
        token = request.headers.get('Authorization')
        if not token or token != f'Token {API_TOKEN}':
            return jsonify({'message': 'Token is missing or invalid!'}), 403
        return f(*args, **kwargs)
    return decorated

@app.route('/api/call_recordings', methods=['GET'])
token_required
def get_call_recordings():
    try:
        # Fetch call recordings from the external API
        response = requests.get(EXTERNAL_API_URL)
        response.raise_for_status()  # Raise an error for bad responses

        # Parse call recordings and funding info
        recordings = response.json()
        call_recordings = []
        funding_info = []

        for recording in recordings:
            call_recordings.append({
                'id': recording['id'],
                'title': recording['title'],
                'duration': recording['duration']
            })
            funding_info.append({
                'recording_id': recording['id'],
                'funding_amount': recording['funding']['amount'],
                'currency': recording['funding']['currency']
            })

        return jsonify({
            'call_recordings': call_recordings,
            'funding_info': funding_info
        }), 200
    except requests.exceptions.RequestException as e:
        return jsonify({'message': str(e)}), 500

if __name__ == '__main__':
    app.run(debug=True)